src/v/compression/include/compression/lz4_decompression_buffers.h

```diff
@@ -100,4 +104,54 @@ class lz4_decompression_buffers {
 std::ostream& operator<<(
   std::ostream&, lz4_decompression_buffers::alloc_ctx::allocation_state);
 
+inline lz4_decompression_buffers::alloc_ctx::allocation_state operator|(
+  lz4_decompression_buffers::alloc_ctx::allocation_state a,
+  lz4_decompression_buffers::alloc_ctx::allocation_state b) {
+    using t = std::underlying_type_t<
+      lz4_decompression_buffers::alloc_ctx::allocation_state>;
+    return lz4_decompression_buffers::alloc_ctx::allocation_state(t(a) | t(b));
+}
+
+inline void operator|=(
+  lz4_decompression_buffers::alloc_ctx::allocation_state& a,
+  lz4_decompression_buffers::alloc_ctx::allocation_state b) {
+    a = (a | b);
+}
+
+inline lz4_decompression_buffers::alloc_ctx::allocation_state operator&(
+  lz4_decompression_buffers::alloc_ctx::allocation_state a,
+  lz4_decompression_buffers::alloc_ctx::allocation_state b) {
+    using t = std::underlying_type_t<
+      lz4_decompression_buffers::alloc_ctx::allocation_state>;
+    return lz4_decompression_buffers::alloc_ctx::allocation_state(t(a) & t(b));
+}
+
+inline void operator&=(
+  lz4_decompression_buffers::alloc_ctx::allocation_state& a,
+  lz4_decompression_buffers::alloc_ctx::allocation_state b) {
+    a = (a & b);
+}
+
+inline lz4_decompression_buffers::alloc_ctx::allocation_state
+operator~(lz4_decompression_buffers::alloc_ctx::allocation_state a) {
+    return lz4_decompression_buffers::alloc_ctx::allocation_state(
+      ~std::underlying_type_t<
+        lz4_decompression_buffers::alloc_ctx::allocation_state>(a));
+}
```

Comment:
Ah, I see we are using them as a bitset<2>. It would be great to mention that as a comment somewhere, or just use a bitset<2>. It's a bit surprising to see bitwise operations on integer types

Reply:
I moved the bitwise operators to an anonymous namespace since they are not used outside of the C api. I will also add a comment to the enum class.